platform/src/main/java/org/hillview/maps/ToCatMap.java

```diff
@@ -0,0 +1,67 @@
+package org.hillview.maps;
+
+import org.hillview.dataset.api.IMap;
+import org.hillview.table.CategoryArrayColumn;
+import org.hillview.table.CategoryListColumn;
+import org.hillview.table.ColumnDescription;
+import org.hillview.table.Table;
+import org.hillview.table.api.ContentsKind;
+import org.hillview.table.api.IColumn;
+import org.hillview.table.api.IRowIterator;
+import org.hillview.table.api.ITable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This map receives a column name of the input table, and returns a table with the same column, with additionally
+ * that specified column converted to a categorical column.
+ */
+public class ToCatMap implements IMap<ITable, ITable> {
+    private final String inputColName;
+    private final String catColName;
+
+    /**
+     * @param inputColName The name of the column that has to be converted to a categorical column.
+     */
+    public ToCatMap(String inputColName) {
+        this.inputColName = inputColName;
+        this.catColName = inputColName + " (Cat.)";
+    }
+
+    @Override
+    public ITable apply(ITable table){
+        List<IColumn> columns = new ArrayList<IColumn>();
+
+        // Copy all existing columns to new table
+        Iterable<IColumn> originalColumns = table.getColumns();
+        int index = -1, i = 0;
+        for (IColumn inputColumn : originalColumns) {
+            if (inputColumn.getName().equals(this.inputColName))
+                index = i;
+            columns.add(inputColumn);
+            i++;
+        }
+
+        // Get the column we want to map to a categorical column
+        IColumn inputColumn = table.getColumn(this.inputColName);
+        if (inputColumn.getDescription().kind == ContentsKind.Category)
```

Comment:
Actually the column must be a String column for this to make sense.

Reply:
It could also make sense for integer columns. For example, the MNIST dataset might have had the labels (digits 0..9) in the table as integers, not categories. In general, every column should be convertible to a string (right?) so that makes it also convertible to a category (as long as the number of unique items is small enough to make sense).

Reply:
Yes, we eventually will have almost all-to-all conversions.
But for numbers this will be done differently - probably by calling Integer.toString().